run_test.go

```diff
@@ -210,14 +210,24 @@ func TestRunsForceCancel(t *testing.T) {
 	t.Run("after a normal cancel", func(t *testing.T) {
 		// Request the normal cancel
 		err := client.Runs.Cancel(ctx, rTest.ID, RunCancelOptions{})
-		assert.NoError(t, err)
+		require.NoError(t, err)
 
-		// Allow time for the async cancel request to operate.
-		time.Sleep(5 * time.Second)
+		for i := 1; ; i++ {
+			// Refresh the view of the run
+			rTest, err = client.Runs.Read(ctx, rTest.ID)
+			require.NoError(t, err)
 
-		// Refresh the view of the run
-		rTest, err = client.Runs.Read(ctx, rTest.ID)
-		assert.NoError(t, err)
+			// Check if the timestamp is present.
+			if !rTest.ForceCancelAvailableAt.IsZero() {
+				break
+			}
```

Comment:
What I think we should do here is `if rTest.Status == tfe.RunCanceled` as that means the async task is completed. The the next test will then confirm that a ForceCancelAvailableAt is properly set now the run is canceled.

Reply:
That's actually not the case. The `canceled` status means that the run successfully canceled and has now ended.

Reply:
Ah... So it will remain in the `RunCanceling` status until it's actually canceled. Makes sense now that I think of it 😏

Reply:
But... shouldn't it then be `if rTest.Status == tfe.RunCanceling` instead? As that is the target status for the next test if I understand it correctly?

Reply:
There's actually not a "canceling" state - the run remains "planning" or "applying" until it reaches a final state such as "canceled". However, once you have initiated a cancel, the API starts returning this timestamp so that you can figure out when a force-cancel would become available, if you need it. Does that make sense?

Reply:
That does make sense. At the same time it makes me wonder if a canceling state wouldn’t actually be helpful to have. But that’s not a question we need to answer here. So all good 👍🏻